## Results & Analysis

### Step 1. Data Preparation
Following links are the galaxy links with uploaded datasets:
- **Uploaded Father_R1_fastq_file:** [Galaxy link](https://usegalaxy.eu/datasets/11ac94870d0bb33a94963102d559fad9/display/?preview=True)
- **Uploaded Father_R2_fastq_file:** [Galaxy link](https://usegalaxy.eu/datasets/11ac94870d0bb33a612b4dc4e32ee64b/display/?preview=True)
- **Uploaded Mother_R1_fastq_file:** [Galaxy link](https://usegalaxy.eu/datasets/11ac94870d0bb33ae00919ce6f0ec993/display/?preview=True)
- **Uploaded Mother_R2_fastq_file:** [Galaxy link](https://usegalaxy.eu/datasets/11ac94870d0bb33af7eada2e86e1cd19/display/?preview=True)
- **Uploaded Proband_R1_fastq_file:** [Galaxy link](https://usegalaxy.eu/datasets/11ac94870d0bb33a331f37d5e50d47ec/display/?preview=True)
- **Uploaded Proband_R2_fastq_file:** [Galaxy link](https://usegalaxy.eu/datasets/11ac94870d0bb33afec1c6ffd1fc3eb1/display/?preview=True)

### Step 2: Quality Control
In this project, after uploading datasets, quality check was performed for using FastQC tool. The FastQC tool gave the output with a good Phred score, which signifies those reads are of good quality. The following are the output of FastQC tool:
- FastQC of Father_R1: [FastQC_webpage](https://usegalaxy.eu/datasets/11ac94870d0bb33a6c560f96aa9380fa/display/?preview=True)
- FastQC of Father_R2: [FastQC_webpage](https://usegalaxy.eu/datasets/11ac94870d0bb33a894719daa729af22/display/?preview=True)
- FastQC of Mother_R1: [FastQC_webpage](https://usegalaxy.eu/datasets/11ac94870d0bb33a48162945b9838974/display/?preview=True)
- FastQC of Mother_R2: [FastQC_webpage](https://usegalaxy.eu/datasets/11ac94870d0bb33ad5c6a2003435e6d1/display/?preview=True)
- FastQC of Proband_R1: [FastQC_webpage](https://usegalaxy.eu/datasets/11ac94870d0bb33a0921102cb8b5fbec/display/?preview=True)
- FastQC of Proband_R2:  [FastQC_webpage](https://usegalaxy.eu/datasets/11ac94870d0bb33adb7b74e87f290b8a/display/?preview=True)

<p align="center">
<br>
<img src="https://user-images.githubusercontent.com/88246395/130108265-eef93927-6cc3-4f58-8dff-b9831c5f984f.JPG" width="300" height="200"> <img src="https://user-images.githubusercontent.com/88246395/130127346-c49aec0f-3c11-4ced-b118-3d58a1f5b404.JPG" width="300" height="200"> <img src="https://user-images.githubusercontent.com/88246395/130127639-9f4dd1d7-4c7b-4735-9789-b39ddf5ed309.JPG" width="300" height="200">
<img src="https://user-images.githubusercontent.com/88246395/130127772-78098874-9e78-4ebf-98d3-71536e146d8e.JPG" width="300" height="200"> <img src="https://user-images.githubusercontent.com/88246395/130128355-e6fadbf6-9681-4bdc-bb55-54a260d3eeb1.JPG" width="300" height="200">  <img src="https://user-images.githubusercontent.com/88246395/130128364-72bb730a-4b8a-488f-a053-aa90be8d6681.JPG" width="300" height="200">
<br>
	<br>
	</p>
	
MultiQC tool was used to aggregate the results from FastQC into a single report. The following is the report generated by MiltiQC tool: [Click Here](https://usegalaxy.eu/datasets/11ac94870d0bb33af4176bff93c8de30/display/?preview=True)
<p align="center">
<br>
<img src="https://user-images.githubusercontent.com/88246395/130129607-b3b1f5a9-ad37-45a9-82e5-383aaa777f9f.JPG" width="700" height="300">
<img src="https://user-images.githubusercontent.com/88246395/130129610-5fd62865-7ef9-4433-8708-81f6e8d9d2dc.JPG" width="700" height="300">
<img src="https://user-images.githubusercontent.com/88246395/130129612-8c1f6219-0807-4033-8a42-e013c6f67a3e.JPG" width="700" height="300">
<br>
	<br>
	</p>

### Step 3: Adapter Trimming
Following are the trimmed datasets:
- Trimmed Father_R1:[Click Here](https://usegalaxy.eu/datasets/11ac94870d0bb33ab83cb16072f2cfd2/display/?preview=True)
- Trimmed Father_R2:[Click Here](https://usegalaxy.eu/datasets/11ac94870d0bb33aa3d66e5af8b06ee8/display/?preview=True)
- Trimmed Mother_R1:[Click Here](https://usegalaxy.eu/datasets/11ac94870d0bb33a4f2a3d32a434c73c/display/?preview=True)
- Trimmed Mother_R2:[Click Here](https://usegalaxy.eu/datasets/11ac94870d0bb33a4f2a3d32a434c73c/display/?preview=True)
- Trimmed Proband_R1:[Click Here](https://usegalaxy.eu/datasets/11ac94870d0bb33a294066e19e357af4/display/?preview=True)
- Trimmed Proband_R2:[Click Here](https://usegalaxy.eu/datasets/11ac94870d0bb33a9f3e07eaab3e8ad7/display/?preview=True)

### Step 4: Read Mapping
Mapping was performed using Map with BWA-MEM tool on fathers, mother’s and proband’s datasets with reference genome. The following is the BAM file geneerated after mapping:
- Mapped Father datasets: [BAM File](https://usegalaxy.eu/datasets/11ac94870d0bb33a97c7cc9f6a20ec8e/display/?preview=True)
- Mapped Mother datasets: [BAM File](https://usegalaxy.eu/datasets/11ac94870d0bb33a29eea5237162eedf/display/?preview=True)
- Mapped Proband datasets: [BAM File](https://usegalaxy.eu/datasets/11ac94870d0bb33a82f28b9314631865/display/?preview=True)
<p align="center">
<br>
<img src="https://user-images.githubusercontent.com/88246395/130130162-9d1e268c-7d2f-41d4-9318-db1ecff964e8.JPG" width="700" height="300">
<br>
	<br>
	</p>

### Step 5: Mapped reads Post-processing
Mapped reads post-processing was performed to filter mapped reads and remove duplicate reads using Filter SAM or BAM, output SAM or BAM and RmDup tools respectively.
- Filtered Father reads: [Click Here]()
- Filtered Mother reads: [Click Here]()
- Filtered Proband reads: [Click Here]()
<br>
- Remove duplicates from Fathers mapped reads: [Click Here]()
- Remove duplicates from Mothers mapped reads: [Click Here]()
- Remove duplicates from Probands mapped reads: [Click Here]()
<br>
<br>

### Step 6:
FreeBayes tool was used to call variants, which specifically finds SNPs, MNPs and Indels. The following is the VCF file output generated by FreeBayes tool:
<p align="center">
<br>
<img src="https://user-images.githubusercontent.com/88246395/130131739-0c1fa7d9-d7d3-40c3-b982-dbea7697d7d9.jpg" width="700" height="300">
<br>
	<br>
	</p>
6. Post-processing FreeBayes calls was performed using bcftools norm, which normalizes indels and split multiallelic sites into multiple rows. 
<br>
<br>
7. For variant annotation, SnpEff eff tool and SnpSift tool were used. The following output was generated by SnpEff tool:
<p align="center">
<br>
<img src="https://user-images.githubusercontent.com/88246395/130132978-f990735e-5d51-420e-9d02-27df96e5bcca.JPG" width="700" height="300">
<img src="https://user-images.githubusercontent.com/88246395/130132988-9418111a-825f-49c3-bc1f-f55dd7d6fd1f.JPG" width="150" height="300"> 
<img src="https://user-images.githubusercontent.com/88246395/130132990-7be4dd60-3015-49f6-b4ec-495f969c042d.JPG" width="550" height="300">
<br>
	<br>
	</p>
8. The combined variant extraction/annotation/storage step was performed on SnpEff-annotated VCF by the GEMINI load tool. The inheritance pattern of the causative variants or candidate variant detection was performed using GEMINI inheritance pattern tool, which gave the following output file: 
<p align="center">
<br>
<img src="https://user-images.githubusercontent.com/88246395/130135237-3a3f08d0-9694-4013-a4f1-1a5af6beb614.JPG" width="700" height="300">
<img src="https://user-images.githubusercontent.com/88246395/130139126-b2633e5e-f77a-4306-9d24-4bde37d769a9.JPG" width="500" height="150">

<br>
	<br>
	</p>
